node/src/components/deploy_acceptor.rs

```diff
@@ -101,120 +184,586 @@ impl DeployAcceptor {
+                    effects.extend(responder.respond(Err(Error::InvalidBlockchain)).ignore());
+                }
+                return effects;
+            }
+        };
+
+        let prestate_hash = *block.state_root_hash();
+        let account_hash = event_metadata.deploy().header().account().to_account_hash();
+        let account_key = account_hash.into();
+
         effect_builder
-            .immediately()
-            .event(move |_| Event::AccountVerificationResult {
-                deploy,
-                source,
-                account_key,
-                verified: Some(true),
-                maybe_responder,
+            .get_account_from_global_state(prestate_hash, account_key)
+            .event(move |maybe_account| Event::GetAccountResult {
+                event_metadata,
+                maybe_account,
+                prestate_hash,
                 verification_start_timestamp,
             })
     }
 
-    #[allow(clippy::too_many_arguments)]
-    fn account_verification<REv: ReactorEventT>(
+    fn handle_get_account_result<REv: ReactorEventT>(
         &mut self,
         effect_builder: EffectBuilder<REv>,
-        deploy: Box<Deploy>,
-        source: Source<NodeId>,
-        account_key: Key,
-        verified: Option<bool>,
-        maybe_responder: Option<Responder<Result<(), Error>>>,
+        event_metadata: EventMetadata,
+        prestate_hash: Digest,
+        maybe_account: Option<Account>,
         verification_start_timestamp: Timestamp,
     ) -> Effects<Event> {
-        let mut effects = Effects::new();
-
-        match verified {
-            Some(true) => {
-                effects.extend(effect_builder.put_deploy_to_storage(deploy.clone()).event(
-                    move |is_new| Event::PutToStorageResult {
-                        deploy,
-                        source,
-                        is_new,
-                        maybe_responder,
+        match maybe_account {
+            None => {
+                let account_hash = event_metadata.deploy().header().account().to_account_hash();
+                effect_builder
+                    .immediately()
+                    .event(move |_| Event::InvalidDeployResult {
+                        event_metadata,
+                        error: Error::InvalidDeployParameters {
+                            prestate_hash,
+                            failure: DeployParameterFailure::NonexistentAccount { account_hash },
+                        },
                         verification_start_timestamp,
+                    })
+            }
+            Some(account) => {
+                let authorization_keys = event_metadata
+                    .deploy()
+                    .approvals()
+                    .iter()
+                    .map(|approval| approval.signer().to_account_hash())
+                    .collect();
+                if !account.can_authorize(&authorization_keys) {
```

Comment:
I think the logic of `can_authorize` is invalid. It requires that _all_ keys that signed the deploy are associated with the account. It should verify that there's enough of them to authorize the deploy execution.

Reply:
@Fraser999 What is your thought here?

Reply:
In conversation with @Fraser999 , the functionality seems valid, and it seems outside of the scope of PR anyway